src/exchndl/exchndl.cpp

```diff
@@ -110,14 +110,11 @@ GenerateExceptionReport(PEXCEPTION_POINTERS pExceptionInfo)
 
     dumpModules(hProcess);
 
-    // TODO: Use GetFileVersionInfo on kernel32.dll as recommended on
-    // https://msdn.microsoft.com/en-us/library/windows/desktop/ms724429.aspx
-    // for Windows 10 detection?
-    OSVERSIONINFO osvi;
-    ZeroMemory(&osvi, sizeof osvi);
-    osvi.dwOSVersionInfoSize = sizeof osvi;
-    GetVersionEx(&osvi);
-    lprintf("Windows %lu.%lu.%lu\n", osvi.dwMajorVersion, osvi.dwMinorVersion, osvi.dwBuildNumber);
+    WORD awVInfo[4];
+    if (getModuleVersionInfo("C:\\Windows\\System32\\kernel32.dll", awVInfo))
```

Comment:
I now use `GetModuleFileName` and `GetModuleHandle` to avoid the string concatenation. The function `getModuleVersionInfo` isn't unicode aware, so this might not work if the system directory contains unicode, but that should be extremely unlikely and would probably break the system in various other places already.